src/main/java/gregicadditions/machines/TileEntityProcessingArray.java

```diff
@@ -419,35 +416,55 @@ protected boolean setupAndConsumeRecipeInputs(Recipe recipe) {
 				recipe.matches(true, importInventory, importFluids);
 		}
 
+		protected boolean didMachinesChange(ItemStack newMachineStack, ItemStack oldMachineStack) {
+
+			return newMachineStack != null && oldMachineStack != null && !ItemStack.areItemStacksEqual(oldMachineStack, newMachineStack);
```

Comment:
I added in the null checks because I want it to fail if the machine stack is removed during the processing of a recipe. In addition, `ItemStack.areItemStacksEqual` checks if the stacks are empty, while I am setting the machine stacks to null to invalidate them.